External/FEXCore/Source/Interface/Core/OpcodeDispatcher/Vector.cpp

```diff
@@ -4620,23 +4620,76 @@ void OpDispatchBuilder::PCMPXSTRXOpImpl(OpcodeArgs, bool IsExplicit) {
   } else {
     IntermediateResult = _VPCMPISTRX(Src1, Src2, Control);
   }
-  OrderedNode *ResultNoFlags = _And(IntermediateResult, _Constant(0xFFFF));
 
-  // For the indexed variant of the instructions, if control[6] is set, then we
-  // store the index of the most significant bit into ECX. If it's not set,
-  // then we store the least significant bit.
   OrderedNode *ZeroConst = _Constant(0);
-
-  const auto ECXResult = [&]() -> OrderedNode* {
+  OrderedNode *UINT16MaxConst = _Constant(0xFFFF);
+  OrderedNode *ResultNoFlags = _And(IntermediateResult, UINT16MaxConst);
+
+  if (IsMask) {
+    // For the masked variant of the instructions, if control[6] is set, then we
+    // need to expand the intermediate result into a byte or word mask (depending
+    // on data size specified in control[1]) along the entire length of XMM0,
+    // where set bits in the intermediate result set the corresponding entry
+    // in XMM0 to all 1s and unset bits set the corresponding entry to all 0s.
+    //
+    // If control[6] is not set, then we just store the intermediate result as-is
+    // into the least significant bits of XMM0 and zero extend it.
+    const auto IsExpandedMask = (Control & 0b0100'0000) != 0;
+
+    if (IsExpandedMask) {
+      // We need to iterate over the intermediate result and expand the mask into XMM0
+      // elements, so we'll need to do a little bit of block shenanigans.
+      const auto ElementSize = 1U  << (Control & 1);
+      const auto NumElements = 16U >> (Control & 1);
+
+      // We're effectively doing the IR equivalent of:
+      //
+      // XMMVector TmpVector = {};
+      // for (int i = 0; i < NumElements; i++) {
+      //    if (((IntermediateResult >> i) & 1) != 0) {
+      //      memcpy(&TmpVector + (i * ElementSize), &UINT16MaxConst, ElementSize);
+      //    }
+      // }
+      // XMM0 = TmpVector;
+      //
+      StoreXMMRegister(0, _VectorZero(Core::CPUState::XMM_SSE_REG_SIZE));
```

Comment:
Can't this entire loop be unrolled and all the block logic removed with a bit of a change to use Sbfe and insert?
```cpp
      OrderedNode *NewXMM0 = _VectorZero(Core::CPUState::XMM_SSE_REG_SIZE);
      for (uint32_t i = 0; i < NumElements; i++) {
        OrderedNode *SignBit = _Sbfe(1, i, ResultNoFlags);
        NewXMM0 = _VInsGPR(Core::CPUState::XMM_SSE_REG_SIZE, ElementSize, i, NewXMM0, SignBit);
      }
      StoreXMMRegister(0, NewXMM0);

Reply:
Oh nice! I didn't even think of Sbfe. Moved over to that instead